autogpt_platform/frontend/src/lib/autogpt-server-api/types.ts

```diff
@@ -65,6 +65,167 @@ export type BlockIOSimpleTypeSubSchema =
   | BlockIOBooleanSubSchema
   | BlockIONullSubSchema;
 
+export enum DataType {
+  SHORT_TEXT = "short-text",
+  LONG_TEXT = "long-text",
+  NUMBER = "number",
+  DATE = "date",
+  TIME = "time",
+  DATE_TIME = "date-time",
+  FILE = "file",
+  SELECT = "select",
+  MULTI_SELECT = "multi-select",
+  TOGGLE = "toggle",
+  CREDENTIALS = "credentials",
+  OBJECT = "object",
+  KEY_VALUE = "key-value",
+  ARRAY = "array",
+}
+
+export function determineDataType(schema: BlockIOSubSchema): DataType {
+  function handleStringSchema(strSchema: BlockIOStringSubSchema): DataType {
+    if ("format" in strSchema && strSchema.format) {
+      switch (strSchema.format) {
+        case "date":
+          return DataType.DATE;
+        case "time":
+          return DataType.TIME;
+        case "date-time":
+          return DataType.DATE_TIME;
+        case "file":
+          return DataType.FILE;
+        case "short-text":
+          return DataType.SHORT_TEXT;
+        case "long-text":
+          return DataType.LONG_TEXT;
+        default:
+          break;
+      }
+    }
+    if (strSchema.enum) return DataType.SELECT;
+    if (strSchema.maxLength && strSchema.maxLength > 200)
+      return DataType.LONG_TEXT;
+    return DataType.SHORT_TEXT;
+  }
+
+  function handleSingleTypeSchema(subSchema: BlockIOSubSchema): DataType {
```

Comment:
this doesn't need to be a nested function

Reply:
moved